### DISCUSSION TITLE
Question about conception

### DISCUSSION THREAD

**Discussion by Hraesvelg** [cookiecutter/cookiecutter-django#4721] [Q&A]
hi, i would like to use this template for a new project but i have some questions about conception : 

Before to ask my question, i read all documentation but i didn't found information...

- the first one is more a "bug" i think, i wonder why the package project_slug.users.api is not a module ? Where is the __init__ file ? 
- second point, why conftest.py and tests packages in general are included in application package ? is it not better to separate fully test package from application 

for example : 

/project
/project/app1/...
/project/app2/...

/tests
/tests/app1/...
/tests/app2/...
/tests/conftest.py

- and last point, of course, i will create multiple applications in my project, where should application be placed ? in the same place that user package ?

thanks.

**Comment by browniebroke** [comment]
1. Probably... Although since Python 3.3 and [PEP 420](https://peps.python.org/pep-0420/) the `__init__.py` isn't strictly required, I've seen some tools not working as expected when it's missing (pytest). Feel free to send a PR to add it, it was probably overlooked.
2. We're following the Django convention here, as you can see on the "Where should the tests live?" section on [the page about writing tests](https://docs.djangoproject.com/en/4.2/topics/testing/overview/). Basically, it's quite common to have each app in a project defining their own tests.
3. Yes, should be next to the `user` package, itself being a Django app.

**Comment by Hraesvelg** [comment]
Thanks for your answer Brownie

1. ok, i will try to pull request the fix
2. if i follow your link and the link under the note : https://docs.djangoproject.com/en/4.2/topics/testing/advanced/#testing-reusable-application. we can see the polls application fully separate to test. ok because we are in template, the standard option can stay to have the module test in the same package but may be with an option -> separate tests package [y/n] ? 
because developer when he create project he know at the beginning if yes or no the project will contains many applications...
3. ok good

another question/idea : 

4. Why template "force" to use django-allauth ? i think template can add several options to choose method to auth ! allauth/LDAP/...
5. Also last point, create template failed if we don't choose cloud provider and if we choose no for whitenoise ! but why ? we can always delegate management for static file to apache/nginx etc etc
